File: DEVBGDEMO/DEVBGDEMO/Pages/LoginPage.cs
using OpenQA.Selenium;
using OpenQA.Selenium.Support.UI;


namespace DEVBGDEMO.Pages
{
    public class LoginPage
    {
        private readonly IWebDriver _driver;
        private IWebElement EmailInput => _driver.FindElement(By.Id("userName"));
        private IWebElement PasswordInput => _driver.FindElement(By.Id("password"));
        private IWebElement SubmitBtn => _driver.FindElement(By.Id("login"));
        private IWebElement DashboardMarker => _driver.FindElement(By.Id("books-wrapper"));

        public LoginPage(IWebDriver driver)
        {
            _driver = driver;
        }

        public void Open(string baseUrl)
        {
            _driver.Navigate().GoToUrl($"{baseUrl}/login");
        }

        public void Login(string email, string password)
        {
            EmailInput.Clear();
            EmailInput.SendKeys(email);
            PasswordInput.Clear();
            PasswordInput.SendKeys(password);
            SubmitBtn.Click();
        }

        public bool IsDashboardVisible()
        {
            var wait = new WebDriverWait(_driver, TimeSpan.FromSeconds(10));
            return wait.Until(_ => DashboardMarker.Displayed);
        }
    }
}

File: DEVBGJAVADEMO/DEVBGJAVADEMO/src/test/java/pages/LoginPage.java
package pages;

import org.openqa.selenium.*;
import org.openqa.selenium.support.ui.WebDriverWait;
import java.time.Duration;

public class LoginPage {
    private final WebDriver driver;
    private By email = By.id("userName");
    private By password = By.id("password");
    private By submit = By.id("login");
    private By dashboard = By.id("books-wrapper");

    public LoginPage(WebDriver driver){
        this.driver = driver;
    }

    public void open(String baseUrl){
        driver.navigate().to(baseUrl + "/login");
    }
    public void login(String e, String p){
        driver.findElement(email).clear(); driver.findElement(email).sendKeys(e);
        driver.findElement(password).clear(); driver.findElement(password).sendKeys(p);
        driver.findElement(submit).click();
    }
    public boolean isDashboardVisible(){
        new WebDriverWait(driver, Duration.ofSeconds(10))
                .until(d -> d.findElement(dashboard).isDisplayed());
        return true;
    }
}
